### ISSUE TITLE
Unit tests with MKL crashing only in Xcode (Mac Big Sur 11.6.1)

### ISSUE THREAD

**Issue by nathanshauer** [labmec/neopz#126]
TL;DR at the end.
In my Mac (Big Sur 11.6.1), three unit tests crash if running them in Xcode with MKL. These crashes do not happen if PZ is compiled with Makefiles. The tests that fail are:

	  1 - TestReduced (Failed)

	  9 - TestMatrix (Failed)

	 23 - TestSBFem (Failed)

Here is the crash message in TestReduced. This happens the first time an MKL pardiso call is made.

<img width="1432" alt="Screen Shot 2021-09-24 at 6 15 34 PM" src="https://user-images.githubusercontent.com/11822201/141334216-9b5ce165-aada-4d19-8dd7-c7250bc923e6.png">

Then, I started wondering why can't MKL be seen by the unit test library? Motivated by the following article:
[CMake: Public VS Private VS Interface](https://leimao.github.io/blog/CMake-Public-Private-Interface/)

I thought that the problem could be related to the fact that MKL is linked to PZ via the cmake PRIVATE tag. See [EnableMKL.cmake](https://github.com/labmec/neopz/blob/50755d7cc2e84d4590e0f05959cbffd31ba6a051/cmake/EnableMKL.cmake#L21)

And indeed, using the PUBLIC tag instead of PRIVATE solved the problem and all the unit tests passed successfully in Xcode.

However, this seems like overkill since we don't need every library that links with PZ to also link with MKL. Additionally, it seems especially weird since things work fine compiling with Makefile instead of Xcode.

Does anyone with a mac also have this issue?

I guess the lazy man solution would be to put, in cmake, an if(Xcode) {link PUBLIC} else {link PRIVATE}

TL;DR:
- Some Unit Tests crash in Xcode with MKL
- All Unit Tests pass successfully compiling with Makefile
- Can be solved by linking MKL with cmake PUBLIC tag instead of PRIVATE
- The fix seems like overkill since only PZ calls MKL functions

**Comment by philippedevloo** [comment]
I am running MacOS Monterey and all tests passed. I noted I needed to build all the targets before running the BuildTest?

**Comment by nathanshauer** [comment]
> I am running MacOS Monterey and all tests passed. I noted I needed to build all the targets before running the BuildTest?

I tried building all the targets once more but it didn't help. I guess the way to go is to update my mac...
